packages/opentelemetry-instrumentation-anthropic/opentelemetry/instrumentation/anthropic/__init__.py

```diff
@@ -152,21 +152,56 @@ async def _aset_input_attributes(span, kwargs):
                 set_span_attribute(
                     span, f"{SpanAttributes.LLM_PROMPTS}.{i}.role", message.get("role")
                 )
+        
+        if kwargs.get("tools") is not None:
+            for i, tool in enumerate(kwargs.get("tools")):
+                prefix = f"{SpanAttributes.LLM_REQUEST_FUNCTIONS}.{i}"
+                set_span_attribute(span, f"{prefix}.name", tool.get("name"))
+                set_span_attribute(span, f"{prefix}.description", tool.get("description"))
+                input_schema = tool.get("input_schema")
+                if input_schema is not None:
+                    set_span_attribute(span, f"{prefix}.input_schema", json.dumps(input_schema))
 
 
 def _set_span_completions(span, response):
     index = 0
     prefix = f"{SpanAttributes.LLM_COMPLETIONS}.{index}"
     set_span_attribute(span, f"{prefix}.finish_reason", response.get("stop_reason"))
+    if response.get("role"):
+        set_span_attribute(span, f"{prefix}.role", response.get("role"))
+
     if response.get("completion"):
         set_span_attribute(span, f"{prefix}.content", response.get("completion"))
     elif response.get("content"):
-        for i, content in enumerate(response.get("content")):
-            set_span_attribute(
-                span,
-                f"{SpanAttributes.LLM_COMPLETIONS}.{i}.content",
-                content.text,
-            )
+        tool_call_index = 0
+        text = ""
+        for content in response.get("content"):
+            content_block_type = content.type
+            # usually, Antrhopic responds with just one text block,
+            # but the API allows for multiple text blocks, so concatenate them
+            if content_block_type == "text":
+                text += content.text
```

Comment:
@nirga looking for your input on this. I tried to stay as close to OpenAI as possible. For OpenAI the `index`, is the ordinal number of the output in the `choices` array. I believe, there is no such thing in Anthropic, so `index` is always 0.

In theory, Anthropic could respond with something like `[text, tool_use, text, text, tool_use]`. I've never seen it do this in practice, but just in case, instead of hard-coding, I decided to concatenate everything in the text block.

An alternative to my approach would be to set these `text` blocks as `gen_ai.completions.i.content`, but I don't think this is consistent with the `choices` in OpenAI.

Reply:
Makes sense @dinmukhamedm! Long term we plan to move these to otel events so there will be a "cleaner" solution